src/Pact/Server/API.hs

```diff
@@ -1,38 +1,256 @@
+pactServerSwagger = toSwagger pactServerAPI
+
+writeSwagger :: FilePath -> Swagger -> IO ()
+writeSwagger fn = BSL8.writeFile fn . encode
+
+
+
+-- ORPHANS for swagger
+
+instance ToSchema (Command Text)
+
+instance ToSchema (CommandResult Hash)
+
+instance ToSchema SubmitBatch
+
+instance ToSchema RequestKeys
+
+instance ToSchema Poll
+
+instance ToSchema PollResponses
+
+instance ToSchema ListenerRequest
+
+instance ToSchema ListenResponse
+
+instance ToSchema Analyze.Request
+
+instance ToSchema Analyze.Response
+
+pactHashSchema :: Schema
+pactHashSchema = withSchema byteBase64url $ fixedLength pactHashLength
+
+instance ToSchema RequestKey where
+  declareNamedSchema = declareGenericSchema pactHashSchema
+
+instance ToSchema Hash where
+  declareNamedSchema = declareGenericSchema pactHashSchema
+
+instance ToSchema (TypedHash 'Blake2b_256) where
+  declareNamedSchema = declareGenericSchema pactHashSchema
+
+instance ToSchema PublicKey where
+  declareNamedSchema = declareGenericString
+
+instance ToSchema UserSig
+
+instance ToSchema TxId
+
+instance ToSchema Gas
+
+instance ToSchema Doc where
+  declareNamedSchema = declareGenericString
+
+instance ToSchema Info where
+  declareNamedSchema = declareGenericString
+
+
+instance ToSchema (ModuleDef t) where
+  declareNamedSchema = declareGenericSchema $
+    (schemaOf $ swaggerType SwaggerObject)
+
+instance ToSchema PactErrorType
+
+instance ToSchema PactError
+
+instance ToSchema StackFrame where
+  declareNamedSchema = declareGenericString
+
+instance ToSchema PactExec
+
+instance ToSchema PactContinuation
+
+instance ToSchema Value where
+  declareNamedSchema = declareGenericSchema $
+    (schemaOf $ swaggerType SwaggerObject)
+
+instance ToSchema PactResult
+
+instance ToSchema Guard
+instance ToSchema PactGuard
+instance ToSchema KeySet
+instance ToSchema KeySetName
+instance ToSchema PactId
+instance ToSchema Name
+instance ToSchema ModuleName
+instance ToSchema NamespaceName
+instance ToSchema ModuleGuard
+instance ToSchema UserGuard where
+  declareNamedSchema = declareGenericString
+
+instance ToSchema PactValue
+
+-- | Adapted from 'Map k v' as a naive instance will cause an infinite loop!!
+-- 2.2 swagger2 compat means not using 'additionalProperties' for now
+instance ToSchema (ObjectMap PactValue) where
+  declareNamedSchema _ = do
+    -- Swagger 2.2 compat, not doing schema ref for pact value
+    -- sref <- declareSchemaRef (Proxy :: Proxy PactValue)
+    return $ NamedSchema (Just "ObjectMap") $
+      (schemaOf $ swaggerType SwaggerObject) -- .
+        -- set additionalProperties (Just $ AdditionalPropertiesSchema sref))
+
+instance ToSchema FieldKey
+
+newtype DummyTime = DummyTime UTCTime deriving (Generic)
+instance ToJSON DummyTime where toJSON (DummyTime t) = encoder timeCodec t
+
+instance ToSchema UTCTime where
+  declareNamedSchema = namedSchema "UTCTime" $ sketchSchema
+    (DummyTime $ mkUTCTime (fromGregorian 1970 01 01) $ fromMicroseconds 0)
```

Comment:
Could we make a note here that this is the POSIX epoch?

Reply:
Nah, it's just for an example to get swagger to stop whining about UTCTime